.github/workflows/retrieval-validation.yaml

```diff
@@ -0,0 +1,69 @@
+name: Retrieval validation
+
+on:
+  push:
+    workflow_dispatch:
+    release:
+      types: [published]
```

Comment:
I should change this to validate on merge to main.. by release time it will be too late 😅

Reply:
I have added on push to main as another condition.